src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useMemo } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoItem } from '../TodoItem';
+
+type Props = {
+  visibleToDos: Todo[];
+  onDelete: (id: number) => Promise<void>;
+  onUpdate: (id: number, updated: Partial<Todo>) => void;
+  tempTodo: Todo | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  visibleToDos,
```

Comment:
Small naming suggestion:

```suggestion
  todos,
```
and use `visibleTodos` as a name instead of `renderToDos`